include/quic/connection.hpp

```diff
@@ -288,6 +274,10 @@ namespace oxen::quic
 
         // holds a mapping of active streams
         std::map<int64_t, std::shared_ptr<Stream>> streams;
+        // datagram "pseudo-stream"
+        std::shared_ptr<DatagramIO> datagrams;
+        // stream "pseudo-stream" to finish off packets
```

Comment:
to finish off packets, yes, but also for ngtcp2 to create any packets it may need us to send like ACKs, retransmits, etc.

Reply:
Would be sufficient to just say "to represent ngtcp2 stream ID -1" though